packages/material-ui/src/Badge/Badge.test.js

```diff
@@ -36,361 +36,140 @@ describe('<Badge />', () => {
     });
   });
 
   describe('prop: showZero', () => {
     it('should default to false', () => {
-      const wrapper = shallow(<Badge badgeContent={0}>{testChildren}</Badge>);
-      assert.strictEqual(
-        wrapper
-          .find('span')
-          .at(1)
-          .hasClass(classes.invisible),
-        true,
-      );
+      const wrapper = mount(<Badge {...defaultProps} badgeContent={0} />);
+      assert.strictEqual(findBadge(wrapper).hasClass(classes.invisible), true);
     });
 
     it('should render without the invisible class when false and badgeContent is not 0', () => {
-      const wrapper = shallow(
-        <Badge badgeContent={10} showZero>
-          {testChildren}
-        </Badge>,
-      );
-
-      assert.strictEqual(
-        wrapper
-          .find('span')
-          .at(1)
-          .hasClass(classes.invisible),
-        false,
-      );
+      const wrapper = mount(<Badge {...defaultProps} badgeContent={10} showZero />);
+      assert.strictEqual(findBadge(wrapper).hasClass(classes.invisible), false);
     });
 
     it('should render without the invisible class when true and badgeContent is 0', () => {
-      const wrapper = shallow(
-        <Badge badgeContent={0} showZero>
-          {testChildren}
-        </Badge>,
-      );
-
-      assert.strictEqual(
-        wrapper
-          .find('span')
-          .at(1)
-          .hasClass(classes.invisible),
-        false,
-      );
+      const wrapper = mount(<Badge {...defaultProps} badgeContent={0} showZero />);
+      assert.strictEqual(findBadge(wrapper).hasClass(classes.invisible), false);
     });
 
     it('should render with the invisible class when false and badgeContent is 0', () => {
-      const wrapper = shallow(
-        <Badge badgeContent={0} showZero={false}>
-          {testChildren}
-        </Badge>,
-      );
+      const wrapper = mount(<Badge {...defaultProps} badgeContent={0} showZero={false} />);
 
-      assert.strictEqual(
-        wrapper
-          .find('span')
-          .at(1)
-          .hasClass(classes.invisible),
-        true,
-      );
+      assert.strictEqual(findBadge(wrapper).hasClass(classes.invisible), true);
     });
   });
 
   describe('prop: variant', () => {
     it('should default to standard', () => {
-      const wrapper = shallow(<Badge badgeContent={10}>{testChildren}</Badge>);
-
-      assert.strictEqual(
-        wrapper
-          .find('span')
-          .at(1)
-          .hasClass(classes.badge),
-        true,
-      );
-      assert.strictEqual(
-        wrapper
-          .find('span')
-          .at(1)
-          .hasClass(classes.dot),
-        false,
-      );
+      const wrapper = mount(<Badge {...defaultProps} badgeContent={10} />);
+      assert.strictEqual(findBadge(wrapper).hasClass(classes.badge), true);
+      assert.strictEqual(findBadge(wrapper).hasClass(classes.dot), false);
     });
 
     it('should render without the standard class when variant="standard"', () => {
```

Comment:
It should render with

Reply:
Alright, let's improve the test descriptions at the same time :)